source/extensions/filters/common/local_ratelimit/provider_singleton.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/type/v3/token_bucket.pb.h"
+#include "envoy/type/v3/token_bucket.pb.validate.h"
+
+#include "source/common/config/subscription_base.h"
+#include "source/common/init/target_impl.h"
+#include "source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace LocalRateLimit {
+
+class RateLimiterProviderSingleton;
+using RateLimiterProviderSingletonSharedPtr = std::shared_ptr<RateLimiterProviderSingleton>;
+class RateLimiterProviderSingleton : public Singleton::Instance {
```

Comment:
Please add a description of this class. Specifically it would also be great to describe the design. It isn't a trivial one, and should be described for future contributors in this space (including lifetime of each of the different components).

Reply:
Added